### PR TITLE
Extend ecutils

### PR DESCRIPTION
**Pull request by oleg-kiviljov**
This PR extends functionality of EcUtils module.

### CODE REVIEW

File: lib/hm_crypto/ec_utils/ec_utils.ex [hunk 1]
```diff
@@ -144,4 +171,59 @@ defmodule HmCrypto.EcUtils do
...
+  """
+
+  @spec public_key_from_private_key(ECPrivateKey.t() | String.t()) :: ec_point
+  def public_key_from_private_key(
+        ECPrivateKey.record(publicKey: public_key, parameters: parameters)
+      ) do
+    {ECPoint.record(point: public_key), parameters}
+  end
+
+  def public_key_from_private_key(private_key) when is_binary(private_key) do
```
Comment (tim2CF) [inline lib/hm_crypto/ec_utils/ec_utils.ex:195]: I'm not sure about naming of this clause. Maybe `public_key_from_private_key_pem` of something similar can be better. PEM is not the only binary/text untyped EC key format, we also are using DER as well.

**Review by ischepin — COMMENTED** [review]
Please fill in PR description. Everything else looks good.

File: lib/hm_crypto.ex [hunk 2]
```diff
@@ -42,10 +42,10 @@ defmodule HmCrypto do
 
   """
 
-  @spec sign!(binary(), rsa_digest_type(), HmCrypto.PublicKey.key()) :: binary()
+  @spec sign!(binary(), digest_type(), HmCrypto.PublicKey.key()) :: binary()
```
Comment (ischepin) [inline lib/hm_crypto.ex:45]: I think the spec is incorrect here; it says 3rd arg is public key but function header says its private key. Obviously you need a private key to sign something. Could you please fix it?

**Review by ischepin — CHANGES_REQUESTED** [review]

File: lib/hm_crypto.ex (hunk 2 above)
Comment (oleg-kiviljov) [reply lib/hm_crypto.ex]: adde PrivateKey module

**Pull request merged**